SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/screens/quiz/QuizQuestionActivity.java

```diff
@@ -125,32 +131,15 @@ protected void onCreate(Bundle savedInstanceState) {
         bottomNavigationView.setSelectedItemId(R.id.practiceTheory);
 
         //add questions to question list via helper method --> help us select question
-        db = new CrushersDataBase(this);
+        CrushersDataBase db = new CrushersDataBase(this);
         questionList = db.getAllQuestions();
-        Collections.shuffle(questionList);
-
-        //Collections for categories and custom question amount quizes
-        categories = new HashSet<>();
+//        quizState = new QuizState(true, questionList, null, scoreNumber);
+        totalQuestions = questionList.size();
         specifcQuestionList = new ArrayList<>();
+        categories = new HashSet<>();
+        this.db = new CrushersDataBase(this);
 
-        //default values are 0, ""
-        questionCountSelected = intent.getIntExtra("OPTION_QUESTIONS", 0);
-        categorySelected = intent.getStringExtra("CATEGORY_SELECTED");
-
-        System.out.println(questionCountSelected);
-        System.out.println(categorySelected);
-
-        //Forms custom quiz with question count from previous screen
-        if(questionCountSelected != 0 || !(categorySelected.equals("")))
-            if(!(categorySelected.equals("No Category")))
-                customQuiz(questionCountSelected, categorySelected);
-            else customQuiz(questionCountSelected, "No Category");
-        else {
-            //else start a random quiz -- No settings selected
-            quizState = new QuizState(true, questionList, null, scoreNumber);
-            totalQuestions = questionList.size();
-            addQuestion(questionList);
-        }
+        customQuiz(numberOfQuestions, category);
```

Comment:
Since if you solely pass them through without handling them they will create an exception since expecially for numberOfQuestions it will pass a 0 if there is nothing passed through and then the custom quiz fails on getting 0 questions

Reply:
> How are you handling quizes then where a person does not select a category or/and select a number of questions 

![image](https://user-images.githubusercontent.com/90762420/167124683-a461d064-8208-44d0-80ad-535943bb482c.png)

When the user starts a Quiz, if a category hasn't been selected, it will default to "No Category". Is there a better way to do this? Yes. Yes, there is xD

Reply:
Perfect yea once these are fixed thats all good then my g